src/llama_stack/apis/vector_io/vector_io.py

```diff
@@ -82,16 +82,32 @@ def model_post_init(self, __context):
         if self.metadata and "chunk_id" in self.metadata:
             self.stored_chunk_id = self.metadata.pop("chunk_id")
 
+    @staticmethod
+    def generate_chunk_id(document_id: str, chunk_text: str, chunk_window: str | None = None) -> str:
+        """
+        Generate a unique chunk ID using a hash of the document ID and chunk text.
+        Then use the first 32 characters of the hash to create a UUID.
+
+        :param document_id: The ID of the document this chunk belongs to.
+        :param chunk_text: The text content of the chunk.
+        :param chunk_window: Optional window identifier for the chunk.
+        :returns: A UUID string generated from the hash of the inputs.
+        """
+        hash_input = f"{document_id}:{chunk_text}".encode()
+        if chunk_window:
+            hash_input += f":{chunk_window}".encode()
+        return str(uuid.UUID(hashlib.sha256(hash_input).hexdigest()[:32]))
+
     @property
     def chunk_id(self) -> str:
```

Comment:
I think the bigger issue is its existence in the API file itself. The API should _never_ have behavior like this at all. It should be data types and method declarations. Not implementation. Can we actually remove the usage of this property itself? You should use immutable datatypes -- something at the top level in the routers, etc. should say "oh there is no chunk id so I will generate one and then I will initialize an immutable object with this chunk id"

Reply:
ok, let me know if the new set of changes makes sense!